library/psa_crypto.c

```diff
@@ -5511,21 +5546,26 @@ static psa_status_t psa_key_agreement_ecdh( const uint8_t *peer_key,
                                             size_t shared_secret_size,
                                             size_t *shared_secret_length )
 {
-    mbedtls_ecp_keypair *their_key = NULL;
+    mbedtls_ecp_keypair their_key;
+    psa_key_slot_t their_key_slot;
```

Comment:
Will change.

Reply:
@ronald-cron-arm changing this actually triggers a "missing field initializers" warning, since the compiler is being told to be pedantic about these. Do you want to accept there being a memset, or do I need to add an initializer macro for `psa_key_slot_t`? (such as the initializers in `crypto_struct/h`)

Reply:
I'm uncomfortable with creating a `psa_key_slot_t` on the stack. We have pending bugs (or missing features) that require adding things to it: a reference count, a mutex. It already has things that you aren't setting such as the policy. A `psa_key_slot_t` that isn't part of the key slot array will not have the usual invariants. It's possible but it makes the code harder to understand. Maybe it would be better to define a substructure for just the key type and representation, and pass this to `load_xxx_representation`?

Reply:
Yes I can see that Clang is complaining about that. In fact the right way to do it (and clang does not complain in that case) is:
`psa_key_slot_t their_key_slot = { };`.

Reply:
Empty initializers are a GCC/Clang extension. `{0}` is standard C but triggers warnings with popular compilers. So we must fill all fields in struct initializers.

Mbed TLS only works on platforms where all-bits-zero is a null pointer and doesn't use floating-point values, so all-bit-zero is a valid way to initialize structures to content that is semantically equivalent to logical zero.

Reply:
So what you're saying @gilles-peskine-arm is that I can leave in the memset?

Reply:
I'm ok with the memset. I'm less ok with the local `psa_key_slot_t`. It isn't a blocker, but it's technical debt.

Reply:
Changed the API for `load_xxx_representation` in order to get rid of the stack-allocated `psa_key_slot_t`. See commit 7f39187.